src/script/script.h

```diff
@@ -412,6 +413,32 @@ bool GetScriptOp(CScriptBase::const_iterator& pc, CScriptBase::const_iterator en
 /** Serialized script, used inside transaction inputs and outputs */
 class CScript : public CScriptBase
 {
+private:
+    inline void AppendDataSize(const size_t size)
+    {
+        if (size < OP_PUSHDATA1) {
+            insert(this->end(), static_cast<value_type>(size));
+        } else if (size <= 0xff) {
+            insert(this->end(), OP_PUSHDATA1);
+            insert(this->end(), static_cast<value_type>(size));
+        } else if (size <= 0xffff) {
+            insert(this->end(), OP_PUSHDATA2);
+            value_type _data[2];
+            WriteLE16(_data, size);
+            insert(this->end(), std::cbegin(_data), std::cend(_data));
+        } else {
+            insert(this->end(), OP_PUSHDATA4);
+            value_type _data[4];
+            WriteLE32(_data, size);
+            insert(this->end(), std::cbegin(_data), std::cend(_data));
```

Comment:
Why `this->` everywhere? Is your plan to change all code you touch to use `this->` for clarity going forward?

Like the `static_cast`s and `cbegin()`/`cend()` though.

Reply:
You're right, we don't have an `end` parameter anymore so I'll revert these back to just `end()`, thanks!